Source/Core/Core/CoreTiming.cpp

```diff
@@ -392,9 +405,10 @@ void Advance()
 {
 	MoveEvents();
 
-	int cyclesExecuted = slicelength - PowerPC::ppcState.downcount;
+	int cyclesExecuted = slicelength - DowncountToCycles(PowerPC::ppcState.downcount);
 	globalTimer += cyclesExecuted;
-	PowerPC::ppcState.downcount = slicelength;
+	lastOCFactor = SConfig::GetInstance().m_OCEnable ? SConfig::GetInstance().m_OCFactor : 1.0f;
+	PowerPC::ppcState.downcount = CyclesToDowncount(slicelength);
```

Comment:
(From discussion on IRC) It's not actually changing the clock rate; the clock rate (CPU_CORE_CLOCK in Dolphin) is actually visible to code through performance counters and timing registers.  This just changes the number of instructions executed per clock cycle (generally referred to as IPC).